libdevcore/db.h

```diff
@@ -70,14 +70,16 @@ class DatabaseFace
     virtual void forEach(std::function<bool(Slice, Slice)> f) const = 0;
 };
 
-DEV_SIMPLE_EXCEPTION(FailedToOpenDB);
```

Comment:
I think the purpose of having different exception classes is being able to handle them differently on the calling side. 

What we're really interested in here is what kind of error was returned from DB - corruption or IO error or not found or other, and not which method was called.

Reply:
I agree here. This is good improvement. However, I'd also like to see how the code would look like if we move the "status" from type to error_info, i.e. having `DatabaseError() << errinfo_status(status)`.

Reply:
After getting rid of `NotFound` exception, I see that we have only one place now, where we need to distinguish DB error type.
So I think I will change the code now the way you suggest here.

Reply:
Let me know when the next review is needed.